devito/ops/transformer.py

```diff
@@ -45,17 +61,61 @@ def opsit(trees, count):
         parameters
     )
 
-    stencil_arrays_initializations = itertools.chain(*[
-        to_ops_stencil(p, node_factory.ops_args_accesses[p])
+    # Extract all symbols that need to be converted to ops_dat
+    to_dat = set()
+    symbols = set(FindSymbols('symbolics').visit(trees[0].root))
+    symbols -= set(FindSymbols('defines').visit(trees[0].root))
+    to_dat |= symbols
+
+    # To ensure deterministic code generation we order the datasets to
+    # be generated (since a set is an unordered collection)
+    to_dat = filter_sorted(to_dat)
```

Comment:
We can use `parameters` here, no need to use FindSymbols

Reply:
This was not created here. It was just moved. Because I need the dats to be here.
stencil and dat are used into ops_par_loop call.

Reply:
This was done in PR #891 about decl_dat, so this change is beyond the scope of this PR.

Reply:
Just using parameters will break, I tried. So to maintain the purpose of this PR I left findSymbols created earlier.

Reply:
Check my PR (#901)

Reply:
The reasoning behind creating the `ops_dat` in the `Operator` instead of the `opsit` function is that we can potentially use the same data in multiple loop nests. If we create the `ops_dat` in `opsit` we could end up recreating datasets that are used in multiple nests insteed of reusing them